src/main/java/io/codelex/flightplanner/FlightInDatabaseService.java

```diff
@@ -0,0 +1,138 @@
+
+        Flight newFlight = new Flight(
+                fromAirport,
+                toAirport,
+                flightRequest.getCarrier(),
+                LocalDateTime.parse(flightRequest.getDepartureTime(), DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")),
+                LocalDateTime.parse(flightRequest.getArrivalTime(), DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")),
+                null
+        );
+
+        Flight savedFlight = flightInDatabaseRepository.save(newFlight);
+
+        return new FlightResponse(
+                fromAirport,
+                toAirport,
+                savedFlight.getCarrier(),
+                savedFlight.getDepartureTime().format(timeFormatter),
+                savedFlight.getArrivalTime().format(timeFormatter),
+                savedFlight.getId()
+        );
+    }
+
+    @Override
+    public void deleteFlight(Integer flightId) {
+        Optional<Flight> existingFlight = flightInDatabaseRepository.findById(flightId);
+        if (existingFlight.isPresent()) {
+            flightInDatabaseRepository.deleteById(flightId);
+        } else {
+            throw new ResponseStatusException(HttpStatus.OK);
+        }
+    }
+
+    @Override
+    public Optional<Flight> getFlightById(Integer id) {
+        return flightInDatabaseRepository.findById(id);
+    }
+
+    @Override
+    public void clearFlights() {
+    flightInDatabaseRepository.deleteAll();
+    airportInDatabaseRepository.deleteAll();
+    }
+
+    @Override
+    public PageResult<Flight> searchFlights(SearchFlightsRequest request) {
+        flightValidator.validateNullFields(request);
+        flightValidator.validateSameAirports(request.getFrom(), request.getTo());
+
+        List<Flight> foundFlights = flightInDatabaseRepository.findAll()
+                .stream()
+                .filter(fl -> isFlightMatchingRequest(fl, request))
+                .toList();
+
+
+        int totalItems = foundFlights.size();
+        int pageSize = 5;
+        int totalPages = (int) Math.ceil((double) totalItems / pageSize);
+
+        List<Flight> paginatedFlights = foundFlights
+                .stream()
+                .limit(pageSize)
+                .collect(Collectors.toList());
+
+        return new PageResult<>(totalItems, totalPages, paginatedFlights);
+    }
+    private boolean isFlightMatchingRequest(Flight flight, SearchFlightsRequest request) {
+        LocalDate requestDepartureDate = LocalDate.parse(request.getDepartureDate());
+        return flight.getFrom().getAirport().equals(request.getFrom())
+                && flight.getTo().getAirport().equals(request.getTo())
+                && flight.getDepartureTime().toLocalDate().equals(requestDepartureDate);
+    }
+
+
+    @Override
+    public Optional<FlightResponse> findFlightById(Integer flightId) {
+        Optional<Flight> flightOptional = flightInDatabaseRepository.findById(flightId);
+
+        return Optional.ofNullable(flightOptional.map(flight -> new FlightResponse(
```

Comment:
Man šķiet, ka to Optional.ofNullable te nevajag vispār, jo flightOptional jau ir Optional pēc tipa... attiecīgi man liekas, ka vajadzētu strādāt arī šādi:

`flightOptional.map(flight -> new FlightResponse(
...
)).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND)));`